examples/publisher_untyped.rs

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: © Contributors to the iceoryx-rs project
+// SPDX-FileContributor: Mathias Kraus
+
+use iceoryx_rs::PublisherBuilder;
+use iceoryx_rs::Runtime;
+
+use bytes::BufMut;
+
+use std::error::Error;
+use std::thread;
+use std::time::Duration;
+
+fn main() -> Result<(), Box<dyn Error>> {
+    Runtime::init("publisher_untyped");
+
+    let publisher = PublisherBuilder::<[u8]>::new("Radar", "FrontLeft", "Counter").create()?;
+
+    let mut counter = 0u32;
+    loop {
+        let sample = if counter % 2 == 1 {
+            // with initialized slice
+            let mut sample = publisher.loan_slice(std::mem::size_of::<u32>())?;
+            sample.as_mut().put_u32(counter);
+            sample
+        } else {
+            // with uninitialized slice
+            let sample = publisher.loan_uninit_slice(std::mem::size_of::<u32>())?;
+            unsafe {
+                let mut sample = sample.assume_init();
+                sample.as_mut().put_u32(counter);
+                sample
+            }
+        };
+        publisher.publish(sample);
```

Comment:
@neilisaac what is your opinion? Maybe a bit awkward to call `assume_init` before the buffer is written but not that bad.